docs/DevGuide/Brigand.md

```diff
@@ -0,0 +1,1827 @@
+`value` static member variable.
+
+\par
+Because of the `type` type alias, integral_constants behave like lazy
+metafunctions returning themselves.  Most lazy metafunctions producing an
+integral_constant will actually inherit from their result, so `value` will be
+directly available without needing to go through the `type` alias.
+
+\remark
+Prefer std::integral_constant, except for the convenience wrapper
+`tmpl::size_t` or when necessary for type equality comparison.
+
+
+\subsubsection list
+
+\par
+An empty struct templated on a parameter pack, with no additional
+functionality.
+\snippet Test_TMPLDocumentation.cpp tmpl::list
+
+\par
+Most metafunctions that operate on lists will work on any struct template.
+
+
+\subsubsection map
+
+\par
+A collection of key-value \ref pair ""s with unique keys.  See the section on
+\ref map_operations "operations on maps" for details.
+\snippet Test_TMPLDocumentation.cpp tmpl::map
+
+\par
+The actual type of a map is implementation-defined, but it has the same
+template parameters as a call to `map` that would produce it.
+
+\warning
+Equivalent maps may have different types, depending on the order their keys are
+stored in internally.
+
+
+\subsubsection pair
+
+\par
+A pair of types, with easy access to each type in the pair.
+\snippet Test_TMPLDocumentation.cpp tmpl::pair
+
+
+\subsubsection set
+
+\par
+An unordered collection of distinct types.  Trying to create a `set`
+with duplicate entries is an error (but \ref set_insert "insert" ignores
+duplicate entries).
+\snippet Test_TMPLDocumentation.cpp tmpl::set
+
+\par
+The actual type of a set is implementation-defined, but it has the same
+template parameters as a call to `set` that would produce it.
+
+\warning
+Equivalent sets may have different types, depending on the order their elements
+are stored in internally.
+
+
+\subsubsection type_
+
+\par
+A struct templated on a single type `T` containing an alias `type` to `T`.
+\snippet Test_TMPLDocumentation.cpp tmpl::type_
+
+\par
+When extracting the type, programmers are encouraged to use \ref type_from to
+make it clear that the `::type` that would otherwise appear is not an
+evaluation of a lazy metafunction.  See \ref always or \ref identity for
+similar functionality that is intended for use as a metafunction.
+
+
+\subsection Constants
+
+\par
+Brigand defines a few concrete types and type aliases.
+
+
+\subsubsection empty_base
+
+\par
+An empty struct used by \ref inherit and \ref inherit_linearly.  Primarily for
+internal use.
+\snippet Test_TMPLDocumentation.cpp tmpl::empty_base
+
+
+\subsubsection empty_sequence
+
+\par
+An empty \ref list.
+\snippet Test_TMPLDocumentation.cpp tmpl::empty_sequence
+
+\remark
+Prefer just writing an empty list.
+
+
+\subsubsection false_type
+
+\par
+An \ref integral_constant representing `false`.  Similar to std::false_type.
+\snippet Test_TMPLDocumentation.cpp tmpl::false_type
+
+\remark
+Prefer std::false_type.
```

Comment:
Is `tmpl::false_type` interchangable with `std::false_type`?   (i.e. is it always possible to use `std::false_type` instead of `tmpl::false_type`?)

Reply:
Except for the fact that they are different types and so will act differently when something cares about type equality, the `std` version is a drop-in replacement for the `tmpl` version.